tests/cgrep_test.py

```diff
@@ -0,0 +1,533 @@
+    expected_results = [('LINKLOCAL', ['FE80::/10'])]
+    ip = '2001:db8::1'
+    results = get_ip_parents(ip, self.db)
+    self.assertNotEquals(results, expected_results)
+
+  # 8.8.8.8 is in GOOGLE_PUBLIC_DNS_ANYCAST which is inside GOOGLE_DNS
+  def test_one_ip_nested(self):
+    expected_results = sorted((('GOOGLE_DNS', ['8.8.8.8/32']),
+                              ('GOOGLE_DNS -> GOOGLE_PUBLIC_DNS_ANYCAST',
+                               ['8.8.8.8/32']),
+                              ('ANY', ['0.0.0.0/0'])))
+    ip = '8.8.8.8'
+    results = sorted(get_ip_parents(ip, self.db))
+    self.assertEquals(results, expected_results)
+
+  # 2001:4860:4860::8844/128 is in GOOGLE_PUBLIC_DNS_ANYCAST which is
+  # inside GOOGLE_DNS
+  def test_one_ipv6_nested(self):
+    expected_results = sorted((('GOOGLE_DNS', ['2001:4860:4860::8844/128']),
+                              ('GOOGLE_DNS -> GOOGLE_PUBLIC_DNS_ANYCAST',
+                               ['2001:4860:4860::8844/128'])))
+    ip = '2001:4860:4860::8844/128'
+    results = sorted(get_ip_parents(ip, self.db))
+    self.assertEquals(results, expected_results)
+
+  # 1.0.0.1 is inside INTERNAL_SERVER, which is inside INTERNAL_SERVERS, which
+  # is inside SERVERS
+  def test_one_ip_multi_nested(self):
+    expected_results = sorted((('INTERNAL_SERVERS -> INTERNAL_SERVER',
+                               ['1.0.0.1/32']),
+                              ('SERVERS -> INTERNAL_SERVER', ['1.0.0.1/32']),
+                              ('SERVERS -> INTERNAL_SERVERS', ['1.0.0.1/32']),
+                              ('SERVERS', ['1.0.0.1/32']),
+                              ('ANY', ['0.0.0.0/0'])))
+
+    ip = '1.0.0.1'
+    results = sorted(get_ip_parents(ip, self.db))
+    self.assertEquals(results, expected_results)
+
+  #
+  # test 'ip in token' (-i -t)
+  #
+  # 8.8.8.8 is inside GOOGLE_DNS
+  def test_ip_in_token(self):
+    expected_results = r'8.8.8.8 is in GOOGLE_DNS'
+    options = Namespace(
+        ip=('8.8.8.8',),
+        token=('GOOGLE_DNS')
+    )
+    results = compare_ip_token(options, self.db)
+    self.assertEquals(results, expected_results)
+
+  # 2001:4860:4860::8844 is inside GOOGLE_DNS
+  def test_ipv6_in_token(self):
+    expected_results = r'2001:4860:4860::8844 is in GOOGLE_DNS'
+    options = Namespace(
+        ip=('2001:4860:4860::8844',),
+        token=('GOOGLE_DNS')
+    )
+    results = compare_ip_token(options, self.db)
+    self.assertEquals(results, expected_results)
+
+  # 69.171.239.12 is not in GOOGLE_DNS
+  def test_ip_in_token_fail(self):
+    expected_results = r'69.171.239.12 is _not_ in GOOGLE_DNS'
+    options = Namespace(
+        ip=('69.171.239.12',),
+        token=('GOOGLE_DNS')
+    )
+    results = compare_ip_token(options, self.db)
+    self.assertEquals(results, expected_results)
+
+  # 2a03:2880:fffe:c:face:b00c:0:35 is not in GOOGLE_DNS
+  def test_ipv6_in_token_fail(self):
+    expected_results = r'2a03:2880:fffe:c:face:b00c:0:35 is _not_ ' \
```

Comment:
Please try not to use \ to join strings.
https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length